src/attributes/codegen.md

```diff
@@ -42,7 +42,110 @@ The *`no_builtins` [attribute]* may be applied at the crate level to disable
 optimizing certain code patterns to invocations of library functions that are
 assumed to exist.
 
+## The `target_feature` attribute
+
+The *`target_feature` [attribute]* may be applied to an [unsafe function] to
+enable code generation of that function for specific platform architecture
+features. It uses the [_MetaListNameValueStr_] syntax with a single key of
+`enable` whose value is a string of comma-separated feature names to enable.
+
+```rust,ignore
+#[target_feature(enable = "avx2")]
+unsafe fn foo_avx2() {}
+```
+
+Each [target architecture] has a set of features that may be enabled. It is an
+error to specify a feature for a target architecture that the crate is not
+being compiled for.
+
+It is [undefined behavior] to call a function that is compiled with a feature
+that is not supported on the current platform the code is running on.
+
+Functions marked with `target_feature` are not inlined into a context that
+does not support the given features. The `#[inline(always)]` attribute may not
+be used with a `target_feature` attribute.
+
+### Available features
+
+The following is a list of the available feature names.
+
+#### `x86` or `x86_64`
+
+Feature     | Implicitly Enables | Description
+------------|--------------------|-------------------
+`aes`       | `sse2`   | [AES] — Advanced Encryption Standard
+`avx`       | `sse4.2` | [AVX] — Advanced Vector Extensions
+`avx2`      | `avx`    | [AVX2] — Advanced Vector Extensions 2
+`bmi1`      |          | [BMI1] — Bit Manipulation Instruction Sets
+`bmi2`      |          | [BMI2] — Bit Manipulation Instruction Sets 2
+`fma`       | `avx`    | [FMA3] — Three-operand fused multiply-add
+`fxsr`      |          | [`fxsave`] and [`fxrstor`] — Save and restore x87 FPU, MMX Technology, and SSE State
+`lzcnt`     |          | [`lzcnt`] — Leading zeros count
+`pclmulqdq` | `sse2`   | [`pclmulqdq`] — Packed carry-less multiplication quadword
+`popcnt`    |          | [`popcnt`] — Count of bits set to 1
+`rdrand`    |          | [`rdrand`] — Read random number
+`rdseed`    |          | [`rdseed`] — Read random seed
+`sha`       | `sse2`   | [SHA] — Secure Hash Algorithm
+`sse`       |          | [SSE] — Streaming SIMD Extensions
+`sse2`      | `sse`    | [SSE2] — Streaming SIMD Extensions 2
+`sse3`      | `sse2`   | [SSE3] — Streaming SIMD Extensions 3
+`sse4.1`    | `sse3`   | [SSE4.1] — Streaming SIMD Extensions 4.1
+`sse4.2`    | `sse4.1` | [SSE4.2] — Streaming SIMD Extensions 4.2
+`ssse3`     | `sse3`   | [SSSE3] — Supplemental Streaming SIMD Extensions 3
+`xsave`     |          | [`xsave`] — Save processor extended states
+`xsavec`    |          | [`xsavec`] — Save processor extended states with compaction
+`xsaveopt`  |          | [`xsaveopt`] — Save processor extended states optimized
+`xsaves`    |          | [`xsaves`] — Save processor extended states supervisor
+
+[AES]: https://en.wikipedia.org/wiki/AES_instruction_set
```

Comment:
I'm okay with leaving these links here instead of at the bottom, but please leave a comment with rationale.